client/app/components/HtmlContent.jsx

```diff
@@ -2,14 +2,14 @@ import React from "react";
 import PropTypes from "prop-types";
 import { sanitize } from "dompurify";
 
-export default function HtmlContent({ children, ...props }) {
+const HtmlContent = React.memo(function HtmlContent({ children, ...props }) {
   return (
     <div
       {...props}
       dangerouslySetInnerHTML={{ __html: sanitize(children) }} // eslint-disable-line react/no-danger
     />
   );
-}
+});
```

Comment:
Have you tried to use `useMemo` instead of `React.memo`? `React.memo` changes behavior of component (due to shallow comparison of props, component _may_ not properly react to props changes):

```jsx
const html = useMemo(() => ({ __html: sanitize(children) }), [children]);

return <div {...props} dangerouslySetInnerHTML={html} />
```

Reply:
Yes, `useMemo` should have the same performance in concept, but not re-rendering the component at all looked better. For the `HtmlContent` component it should be totally dependent on its props. I quickly checked its usages, it has only an string for className (basic variable types don't create issues with shallow comparison) and the `children` node didn't seem a problem as well.

Reply:
Thanks for explanation! While I personally prefer `useMemo`, `React.memo` looks good as well :+1: